client/src/pages/Actions.ts

```diff
@@ -30,3 +88,41 @@ export const loginUser = (): IdentityToken | null => {
     });
   return identityToken;
 };
+
+export const getGeoLocation = (
+  successCallback: (position: GeoLocation) => void,
+  errorCallback?: (error: any) => void
+) => {
+  if (isWeb) {
+    // Use browser navigator
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(
+        successCallback,
+        errorCallback ? errorCallback : () => {}
+      );
+    }
+  } else {
+    // Use microapps API
+    microapps
+      .getCurrentLocation()
+      .then((loc: any) => Object.assign(emptyGeoLocation(), { coords: loc }))
+      .then((pos: GeoLocation) => successCallback(pos))
+      .catch((err: any) => (errorCallback ? errorCallback(err) : null));
```

Comment:
Assign a default errCallback as errorCalback= (error:any)=>{} and call it here directly
```suggestion
      .catch((err: any) => errorCallback(err));
```

Reply:
I've previously tried setting a default implementation for errorCallback but was getting a typescript error:

> Cannot invoke an object which is possibly 'undefined'.  TS2722

with:
```
if (!errorCallback){
  errorCallback = (err: any) => {};
}
```
Which was strange...